src/core/ActionAtomistic.cpp

```diff
@@ -77,7 +77,8 @@ void ActionAtomistic::requestAtoms(const vector<AtomNumber> & a) {
 // only real atoms are requested to lower level Atoms class
     else unique.insert(indexes[i]);
   }
-
+  updateUniqueLocal();
+  atoms.unique.clear();
```

Comment:
the idea is that clearing unique at every time step is much slower then just updating it, so it is cleared only upon domain decomposition update, neighbour list update, replica-exchange request. In this way there could be cases in which the number of unique atoms is higher than those actually required but is never less than that. In all cases I tested this is faster, it could be slower if we bias very few atoms every step and monitor all the atoms more frequently than neighbour list update.